eslint.config.js

```diff
@@ -73,11 +73,7 @@ const allFilesConfig = tsEslint.config({
 export function getTypeScriptProjectRules(root) {
   return tsEslint.config({
     files: ["**/packages/*/src/**/*.ts", "**/packages/*/src/**/*.tsx"],
-    ignores: [
-      "**/packages/http-client-csharp/**/*",
-      "**/packages/http-client-java/**/*",
-      "**/packages/http-client-python/**/*",
-    ], // Ignore isolated modules
+    ignores: ["**/packages/http-client-csharp/**/*", "**/packages/http-client-python/**/*"], // Ignore isolated modules
```

Comment:
I think I used to disable those because at the time typescript eslint wasn't liking your different tsc engine but that seems to have been resolved if the CI passed

Reply:
Yeah, last year when we enable this, we see some error about `RangeError: Maximum call stack size exceeded`
https://github.com/microsoft/typespec/pull/4670#discussion_r1794836397
(we had an eslint 8 in another repo that can still report a deprecated usage, so we would still handle them if found there)

I've hoped newer version of eslint solved this, and try again here.

The CI looks pass, but in a commit I tried to use a deprecated property from TCGC but we still pass the CI. https://github.com/microsoft/typespec/pull/6364/commits/40d17c8d6d6c769d0fcaa5fc2f30cc0fce67063a
I would need some check on our script. Maybe it forget to call lint...

Reply:
I wouldn’t run lint again in your pipeline it should already run globally in the lint GitHub action.

Reply:
I see.

I think the reason was that ts files in emitter (python/csharp/java) is under `emitter/src`, so we need to add this pattern to `files` https://github.com/microsoft/typespec/pull/6364/commits/df9ad6ffca0a3c7f40b9b47a1974d661a0297bbb